apps/mobile/src/screens/auth/auth.tsx

```diff
@@ -18,31 +18,58 @@ import {
 	useAppSelector,
 	useCallback,
 	useEffect,
+	useNavigation,
 } from "~/libs/hooks/hooks";
 import { globalStyles } from "~/libs/styles/styles";
 import {
+	type NativeStackNavigationProp,
+	type RootNavigationParameterList,
+} from "~/libs/types/types";
+import {
+	type EmailDto,
+	type ResetPasswordDto,
 	type UserSignInRequestDto,
 	type UserSignUpRequestDto,
 } from "~/packages/users/users";
 import { actions as authActions } from "~/slices/auth/auth";
 
-import { SignInForm, SignUpForm } from "./components/components";
+import {
+	ForgotPasswordForm,
+	ResetPasswordForm,
+	SignInForm,
+	SignUpForm,
+} from "./libs/components/components";
 import { styles } from "./styles";
 
 const IOS_KEYBOARD_OFFSET = 40;
 const ANDROID_KEYBOARD_OFFSET = 0;
 
 const Auth: React.FC = () => {
-	const { name } = useAppRoute();
+	const { name, params } = useAppRoute();
+
 	const dispatch = useAppDispatch();
 	const { dataStatus, user } = useAppSelector((state) => state.auth);
 
+	const navigation =
+		useNavigation<NativeStackNavigationProp<RootNavigationParameterList>>();
+
+	const { token } =
+		name === RootScreenName.RESET_PASSWORD
+			? (params as { token: string })
+			: { token: "" };
```

Comment:
Can we simplify it to this?
```suggestion
	const { token: resetPasswordToken = "" } = params ?? {};
```